reconcile/acs_notifiers.py

```diff
@@ -0,0 +1,157 @@
+            custom_fields["security"] = {"id": jira_board.issue_security_id}
+        if escalation_policy.channels.jira_component:
+            custom_fields["components"] = [
+                {"name": escalation_policy.channels.jira_component}
+            ]
+        if escalation_policy.channels.jira_labels:
+            custom_fields["labels"] = escalation_policy.channels.jira_labels
+
+        item = JiraNotifier(
+            name=f"jira-{escalation_policy.name}",
+            board=jira_board.name,
+            url=jira_board.server.server_url,
+            issue_type=jira_board.issue_type or "Task",
+            severity_priority_mappings=sorted(
+                [
+                    SeverityPriorityMapping(**vars(sp))
+                    for sp in jira_board.severity_priority_mappings.mappings
+                ],
+                key=lambda m: m.severity,
+            ),
+            custom_fields=custom_fields if custom_fields else {},
+        )
+
+        return item
+
+    def get_desired_state(self) -> list[JiraNotifier]:
+        return [self._build_jira_notifier(ep) for ep in self._get_escalation_policies()]
+
+    def get_jira_credentials(
+        self, secret_reader: SecretReaderBase
+    ) -> dict[str, JiraCredentials]:
+        return {
+            server.server_url: JiraCredentials(
+                url=server.server_url,
+                username=server.username,
+                token=secret_reader.read_secret(server.token),
+            )
+            for server in query_jira_servers(query_func=self.gql_api.query).jira_servers
+            or []
+        }
+
+    def reconcile(
+        self,
+        desired_state: list[JiraNotifier],
+        current_state: list[JiraNotifier],
+        acs_api: AcsNotifiersApi,
+        jira_credentials: dict[str, JiraCredentials],
+        dry_run: bool,
+    ) -> None:
+        diff = diff_iterables(
+            current=current_state, desired=desired_state, key=lambda x: x.name
+        )
+        for a in diff.add.values():
+            logging.info(f"Create Jira notifier: {a.name}")
+            if not dry_run:
+                acs_api.create_jira_notifier(
+                    a,
+                    jira_credentials=jira_credentials[a.url],
+                )
+        for c in diff.change.values():
+            logging.info(f"Update Jira notifier: {c.desired.name}")
+            if not dry_run:
+                acs_api.update_jira_notifier(
+                    c.desired,
+                    jira_credentials=jira_credentials[c.desired.url],
+                )
+        for d in diff.delete.values():
+            logging.info(f"Delete Jira notifier: {d.name}")
+            if not dry_run:
+                acs_api.delete_jira_notifier(d)
+
+    def run(
+        self,
+        dry_run: bool,
+    ) -> None:
+        vault_settings = get_app_interface_vault_settings()
+        secret_reader = create_secret_reader(use_vault=vault_settings.vault)
```

Comment:
not needed, just use `self.secret_reader`

Reply:
custom_fields=custom_fields if custom_fields else {}

Reply:
did more of the same for existing integrations in #4263